build/azure-pipelines/sql-product-build.yml

```diff
@@ -4,87 +4,136 @@ resources:
     image: sqltoolscontainers.azurecr.io/linux-build-agent:6
     endpoint: SqlToolsContainers
 
-jobs:
-- job: Compile
-  pool:
-    vmImage: 'Ubuntu-20.04'
-  container: linux-x64
-  steps:
-  - script: |
-      set -e
-      echo "##vso[build.addbuildtag]$(VSCODE_QUALITY)"
-    displayName: Add Quality Build Tag
-  - template: sql-product-compile.yml
-  timeoutInMinutes: 120
+stages:
+  - stage: Compile
+    jobs:
+    - job: Compile
+      pool:
+        vmImage: 'Ubuntu-20.04'
+      container: linux-x64
+      steps:
+      - script: |
+          set -e
+          echo "##vso[build.addbuildtag]$(VSCODE_QUALITY)"
+        displayName: Add Quality Build Tag
+      - template: sql-product-compile.yml
+      timeoutInMinutes: 120
 
-- job: macOS
-  condition: and(succeeded(), eq(variables['VSCODE_BUILD_MACOS'], 'true'), ne(variables['VSCODE_QUALITY'], 'saw'))
-  pool:
-    vmImage: 'macos-latest'
-  dependsOn:
-  - Compile
-  steps:
-  - template: darwin/sql-product-build-darwin.yml
-  timeoutInMinutes: 90
+  - stage: macOS
+    condition: and(succeeded(), eq(variables['VSCODE_BUILD_MACOS'], 'true'), ne(variables['VSCODE_QUALITY'], 'saw'))
+    pool:
+      vmImage: 'macos-latest'
+    dependsOn:
+      - Compile
+    jobs:
```

Comment:
as we chatted offline, the matrix strategy is not very helpful for our scenario since the dependencies are different for different arch. I am going to leave it as is.